cairo_zero/tests/src/kakarot/test_kakarot.py

```diff
@@ -142,11 +142,168 @@ def test_should_assert_only_owner(self, cairo_run):
+                    "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+                )
+                + 1,
+                value=0x101,
+            )
+
+        @SyscallHandler.patch("Ownable_owner", SyscallHandler.caller_address)
+        @SyscallHandler.patch(
+            get_storage_var_address(
+                "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+            ),
+            value=1,
+        )
+        @SyscallHandler.patch(
+            get_storage_var_address(
+                "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+            )
+            + 1,
+            value=0x101,
+        )
+        @patch.object(SyscallHandler, "block_number", 0x102)
+        def test_set_base_fee_should_overwrite_current_block_fee_if_next_block_is_applicable(
+            self, cairo_run
+        ):
+            # Because block_number == 102, the mocked 'next_block', available
+            # since block 101, should be moved to 'current_block'.
+            cairo_run("test__set_base_fee", base_fee=2)
+            SyscallHandler.mock_storage.assert_any_call(
+                address=get_storage_var_address(
+                    "Kakarot_base_fee", int.from_bytes(b"current_block", "big")
+                ),
+                value=1,
+            )
+            SyscallHandler.mock_storage.assert_any_call(
+                address=get_storage_var_address(
+                    "Kakarot_base_fee", int.from_bytes(b"current_block", "big")
+                )
+                + 1,
+                value=0x101,
+            )
+            SyscallHandler.mock_storage.assert_any_call(
+                address=get_storage_var_address(
+                    "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+                ),
+                value=2,
+            )
+            SyscallHandler.mock_storage.assert_any_call(
+                address=get_storage_var_address(
+                    "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+                )
+                + 1,
+                value=0x103,
+            )
+
+        @SyscallHandler.patch("Ownable_owner", SyscallHandler.caller_address)
+        @SyscallHandler.patch(
+            get_storage_var_address(
+                "Kakarot_base_fee", int.from_bytes(b"current_block", "big")
+            ),
+            value=1,
+        )
+        @SyscallHandler.patch(
+            get_storage_var_address(
+                "Kakarot_base_fee", int.from_bytes(b"current_block", "big")
+            )
+            + 1,
+            value=0x100,
+        )
+        @SyscallHandler.patch(
+            get_storage_var_address(
+                "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+            ),
+            value=2,
+        )
+        @SyscallHandler.patch(
+            get_storage_var_address(
+                "Kakarot_base_fee", int.from_bytes(b"next_block", "big")
+            )
+            + 1,
+            value=0x101,
+        )
```

Comment:
I'd rather update the patch decorator to handle a list like

```
@patch("Kakarot_base_fee", "current_block", [1, 0x100])
```

and update here

```
cls.patches[selector_if_storage] = value
```

to something like

```
if isinstance(value, Iterable):
    for i, v in enumerate(value):
        cls.patches[selector_if_storage + i] = v
else:
    cls.patches[selector_if_storage] = value
```

Reply:
yeah agreed will get on that